### PR TITLE
Content changes for opposite and same sex (Sweden)

### PR DESCRIPTION
**Pull request by ikennaokpala**
Trello Story: https://trello.com/c/MaF6tohO/104-marriage-abroad-sweden

## Factcheck
* Opposite sex outcomes where user is resident in the UK:
   * [Preview Link](https://smart-answers-pr-2453.herokuapp.com/marriage-abroad/y/sweden/uk/partner_british/opposite_sex)
   * [Link on GOV.UK](https://gov.uk/marriage-abroad/y/sweden/uk/partner_british/opposite_sex)

## Expected changes
* Content changes for opposite outcomes where user is resident in the UK
  * Changes to Legalisation and translation

### Before
screen shot 2016-04-11 at 16 05 13

### After

screen shot 2016-04-11 at 16 04 47

## Factcheck

* Opposite sex marriage outcomes for a resident in Sweden marrying a British/Swedish/National of another country
    * [Preview Link](https://smart-answers-pr-2453.herokuapp.com/marriage-abroad/y/sweden/ceremony_country/partner_british/opposite_sex)
    * [Link on GOV.UK](https://gov.uk/marriage-abroad/y/sweden/ceremony_country/partner_british/opposite_sex)

## Expected changes
* Mostly new content for opposite outcomes where user is resident in Sweden marrying a British/Swedish/National of another country

### Before

screen shot 2016-04-11 at 16 08 50

### After
screen shot 2016-04-11 at 16 08 39

## Factcheck

* Same sex marriage outcomes for a resident in Sweden marrying a British/Swedish/National of another country
    * [Preview Link](https://smart-answers-pr-2453.herokuapp.com/marriage-abroad/y/sweden/ceremony_country/partner_british/opposite_sex)
    * [Link on GOV.UK](https://gov.uk/marriage-abroad/y/sweden/ceremony_country/partner_british/opposite_sex)

## Expected changes
* Mostly new content for same sex outcomes where user is resident in Sweden marrying a British/Swedish/National of another country

### Before

screen shot 2016-04-11 at 16 13 01

### After
screen shot 2016-04-11 at 16 13 33

### CODE REVIEW

**Comment by ikennaokpala** [comment]
FCO are happy with the changes and now ready for review

**Comment by ikennaokpala** [comment]
FCO are happy with the changes and now ready for review

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb [hunk 1]
```diff
@@ -63,6 +66,17 @@
 
     <%= render partial: 'you_will_be_asked_for_cni.govspeak.erb' %>
 
+  <% elsif calculator.resident_of_ceremony_country? && calculator.ceremony_country == 'sweden' %>
+    If you aren’t registered in the Swedish population register you’ll need to provide proof of marital status. A certificate of no impediment (CNI) will be required to prove you’re allowed to marry.
+
+    If you aren’t registered in Sweden, you’ll also need to bring your partner’s extract from the population register as proof of address, or proof of address from the country that you’re living in.
+
+  <% elsif calculator.resident_of_uk? && calculator.ceremony_country == 'bulgaria' %>
```
Comment (chrisroos) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb]: It doesn't look as though these changes belong in this commit.

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb [hunk 2]
```diff
@@ -226,6 +244,21 @@
...
+
+      You can download and fill in (but not sign) the forms in advance.
+
+      ###Posting notice at the embassy
+
+      [Book an appointment for a CNI in Stockholm](https://www.consular-appointments.service.gov.uk/fco/#!/british-embassy-stockholm/certificate-of-no-impediment/slot_picker).
+
+      To give notice of intended marriage at the embassy, you can swear or affirm an oath. If you would like to swear an oath on a holy book (religious), you’ll need to bring your own holy book.
+
+    <% elsif calculator.ceremony_country == 'bulgaria' %>
```
Comment (chrisroos) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb]: As above, it's not clear that these changes should be part of this commit.

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb [hunk 3]
```diff
@@ -362,6 +404,23 @@
...
+      If a notary public witnesses your documents, you must post the signed forms and any supporting documents to the British embassy in Stockholm afterwards, along with the [credit card payment authorisation form](/government/publications/creditdebit-card-payment-authorisation-slip-sweden) to:
+
+      $A
+      British Embassy
+      Consular Section
+      Box 27819
+      115 93 Stockholm
+      $A
+
+    <% elsif calculator.ceremony_country == 'bulgaria' %>
```
Comment (chrisroos) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb]: As above, it's not clear that these changes should be part of this commit.

**Comment by chrisroos** [comment]
If I understand the Trello card / Google doc with changes then it looks like this branch has updated some results that it shouldn't have.

Requested changes (from Google Doc):

A. opposite-sex where user is resident of uk

* /marriage-abroad/y/sweden/uk/partner_british/opposite_sex
* /marriage-abroad/y/sweden/uk/partner_local/opposite_sex
* /marriage-abroad/y/sweden/uk/partner_other/opposite_sex

B. Opposite sex and same sex marriage outcomes for a resident in Sweden marrying a British/Swedish/National of another country.

* /marriage-abroad/y/sweden/ceremony_country/partner_british/opposite_sex
* /marriage-abroad/y/sweden/ceremony_country/partner_local/opposite_sex
* /marriage-abroad/y/sweden/ceremony_country/partner_other/opposite_sex
* /marriage-abroad/y/sweden/ceremony_country/partner_british/same_sex
* /marriage-abroad/y/sweden/ceremony_country/partner_local/same_sex
* /marriage-abroad/y/sweden/ceremony_country/partner_other/same_sex

Actual changes (based on changes in regression test artefacts):

* /marriage-abroad/sweden/ceremony_country/partner_british/opposite_sex.txt
* /marriage-abroad/sweden/ceremony_country/partner_british/same_sex.txt
* /marriage-abroad/sweden/ceremony_country/partner_local/opposite_sex.txt
* /marriage-abroad/sweden/ceremony_country/partner_local/same_sex.txt
* /marriage-abroad/sweden/ceremony_country/partner_other/opposite_sex.txt
* /marriage-abroad/sweden/ceremony_country/partner_other/same_sex.txt
* /marriage-abroad/sweden/third_country/partner_british/same_sex.txt
* /marriage-abroad/sweden/third_country/partner_local/same_sex.txt
* /marriage-abroad/sweden/third_country/partner_other/same_sex.txt
* /marriage-abroad/sweden/uk/partner_british/opposite_sex.txt
* /marriage-abroad/sweden/uk/partner_british/same_sex.txt
* /marriage-abroad/sweden/uk/partner_local/opposite_sex.txt
* /marriage-abroad/sweden/uk/partner_local/same_sex.txt
* /marriage-abroad/sweden/uk/partner_other/opposite_sex.txt
* /marriage-abroad/sweden/uk/partner_other/same_sex.txt

It looks as though these changes from the list above might be unintentional:

* /marriage-abroad/sweden/third_country/partner_british/same_sex.txt
* /marriage-abroad/sweden/third_country/partner_local/same_sex.txt
* /marriage-abroad/sweden/third_country/partner_other/same_sex.txt
* /marriage-abroad/sweden/uk/partner_british/same_sex.txt
* /marriage-abroad/sweden/uk/partner_local/same_sex.txt
* /marriage-abroad/sweden/uk/partner_other/same_sex.txt

**Comment by chrisroos** [comment]
I'm not entirely sure that it makes sense to reuse `outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country` for same sex ceremonies in Sweden, particularly given the problems (mentioned in the previous commit) it appears to have introduced.

I _think_ a better solution might be to introduce a new outcome for all ceremonies in Sweden where the user is resident in Sweden (`outcome_ceremonies_in_sweden_when_residing_in_sweden`). You could route to this outcome by adding a check for this condition above [`elsif calculator.partner_is_opposite_sex?` in marriage-abroad.rb](https://github.com/alphagov/smart-answers/blob/master/lib/smart_answer_flows/marriage-abroad.rb#L118). You'd obviously still need to make the Sweden specific changes to the `outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country` outcome for opposite sex marriages where the user is a resident of the UK.

Let me know if you have any questions about my suggestion.

**Comment by chrisroos** [comment]
This branch should also update the responses-and-expected-results file for marriage-abroad (using `rails r script/generate-responses-and-expected-results-for-smart-answer.rb marriage-abroad`). This is because some ceremonies in Sweden now lead to different outcomes.

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb (hunk 1 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb]: This it is not intentional.. I worked on this PR after working on bulgarian..

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb (hunk 2 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb]: This it is not intentional.. I worked on this PR after working on Bulgaria..

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb (hunk 3 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb]: This it is not intentional.. I worked on this PR after working on Bulgaria..

**Comment by ikennaokpala** [comment]
@chrisroos thanks for picking up on the Bulgarian commit mixup.. 

Your proposed changes suggest changes for opposite sex.. I take it you mean same sex??

**Comment by chrisroos** [comment]
@ikennaokpala: I've looked at your changes and there still appear to be some differences between the results and the text in the Google Doc.

According to the Google Doc, the outcomes for opposite and same sex ceremonies in Sweden should be identical when the user is a resident of Sweden. With the current implementation there are still a few differences:

1. The sentence starting "To get a CNI" includes "or in front of a notary public." This isn't in the Google Doc. Should it be removed?

  * [ceremony_country/partner_british/opposite_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_british/opposite_sex.txt#L13)
  * [ceremony_country/partner_british/same_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_british/same_sex.txt#L13)
  * [ceremony_country/partner_local/opposite_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_local/opposite_sex.txt#L13)
  * [ceremony_country/partner_local/same_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_local/same_sex.txt#L13)
  * [ceremony_country/partner_other/opposite_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_other/opposite_sex.txt#L13)
  * [ceremony_country/partner_other/same_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_other/same_sex.txt#L13)

2. Opposite and same sex outcomes when your partner is "local" or "other" currently include a naturalisation section. This section doesn't appear in the Google Doc. Should it be removed?

  * [ceremony_country/partner_local/opposite_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_local/opposite_sex.txt#L66)
  * [ceremony_country/partner_local/same_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_local/same_sex.txt#L66)
  * [ceremony_country/partner_other/opposite_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_other/opposite_sex.txt#L66)
  * [ceremony_country/partner_other/same_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_other/same_sex.txt#L66)

3. The same sex outcomes are missing the "Receiving a notice of marriage" fee.

  * [ceremony_country/partner_british/same_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_british/same_sex.txt#L69)
  * [ceremony_country/partner_local/same_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_local/same_sex.txt#L73)
  * [ceremony_country/partner_other/same_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_other/same_sex.txt#L73)

  _The opposite sex outcomes include this fee_:

  * [ceremony_country/partner_british/opposite_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_british/opposite_sex.txt#L70)
  * [ceremony_country/partner_local/opposite_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_local/opposite_sex.txt#L74)
  * [ceremony_country/partner_other/opposite_sex](https://github.com/alphagov/smart-answers/blob/marriage-abroad-content-changes-sweden/test/artefacts/marriage-abroad/sweden/ceremony_country/partner_other/opposite_sex.txt#L74)

**Comment by chrisroos** [comment]
I'm still not very keen on reusing `outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country` for all ceremonies in Sweden when resident of the UK. My preference would be to create a new outcome. I've demonstrated this approach in PR #2461 as I think it's easier to see what I mean in code.

**Comment by ikennaokpala** [comment]
@chrisroos.. much appreciated with your efforts and suggestions from PR #2461.

Yes the google doc does not contain two other rounds of changes.. I will be chatting with SI to properly match the differences

**Pull request closed without merging**